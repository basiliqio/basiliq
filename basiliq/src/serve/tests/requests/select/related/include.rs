use super::*;

crate::run_test_request!(
    include_rel_m2o,
    Method::GET,
    format!(
        "/public__articles/{}/public__peoples?include=public__favorite_color",
        ARTICLES_IDS[0]
    ),
    200
);

crate::run_test_request!(
    include_self,
    Method::GET,
    format!(
        "/public__peoples/{}/public__articles?include=public__peoples",
        PEOPLES_IDS[0]
    ),
    200
);

crate::run_test_request!(
    include_rel_o2m,
    Method::GET,
    format!(
        "/public__peoples/{}/public__articles?include=public__people-article",
        PEOPLES_IDS[0]
    ),
    200
);

crate::run_test_request!(
    include_distant,
    Method::GET,
    format!(
        "/public__peoples/{}/public__articles?include=public__peoples.public__comments",
        PEOPLES_IDS[0]
    ),
    200
);
crate::run_test_request!(
    include_distant_self,
    Method::GET,
    format!(
        "/public__peoples/{}/public__articles?include=public__peoples.public__comments.public__articles",
        PEOPLES_IDS[0]
    ),
    200
);
