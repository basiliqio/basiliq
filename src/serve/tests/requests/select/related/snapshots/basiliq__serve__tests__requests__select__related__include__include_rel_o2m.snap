---
source: basiliq/src/serve/tests/requests/select/related/include.rs
expression: res

---
{
  "jsonapi": {
    "version": "1.0"
  },
  "data": [
    {
      "type": "public__articles",
      "id": "fdf715dd-8772-498c-8196-6f4ccb64edef",
      "attributes": {
        "body": "Yes",
        "title": "How to dead"
      },
      "relationships": {
        "public__people-article": {
          "data": [
            {
              "type": "public__people-article",
              "id": "f77b0017-1dfd-4ea1-b036-b228289f89e0"
            },
            {
              "type": "public__people-article",
              "id": "538d2b60-cb73-4b7d-bc39-cceac6abecb8"
            }
          ]
        }
      }
    },
    {
      "type": "public__articles",
      "id": "2dbf5d1a-b029-4456-af6b-339c75b1089c",
      "attributes": {
        "body": "Yeah I know ! Right ?!",
        "title": "Oh my g**"
      },
      "relationships": {
        "public__people-article": {
          "data": {
            "type": "public__people-article",
            "id": "f0523b1e-30df-418d-83dc-876e4c78cfa9"
          }
        }
      }
    }
  ],
  "included": [
    {
      "type": "public__people-article",
      "id": "538d2b60-cb73-4b7d-bc39-cceac6abecb8"
    },
    {
      "type": "public__people-article",
      "id": "f0523b1e-30df-418d-83dc-876e4c78cfa9"
    },
    {
      "type": "public__people-article",
      "id": "f77b0017-1dfd-4ea1-b036-b228289f89e0"
    }
  ]
}
